src/server/utils/utils.js

```diff
@@ -38,3 +38,36 @@ export const isWhitelisted = (key) => {
   const lowerCasedKey = (typeof key === 'string') ? key.toLowerCase() : key;
   return lowerCasedWhitelist.includes(lowerCasedKey);
 };
+
+
+export const fromFieldsToSource = (parsedInfo) => {
```

Comment:
Could you add a simple comment about what it does? like function `transferSlashStyleToDashStyle`'s comment does: what's the input and what'll be the output. 
And since there's a pretty fancy traversal algorithm inside, we should definitely add unit test for this function. Even a simplest test would be great! thanks!